src/components/ButtonTwoLines/index.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from "react"
+import { Icon, Stack, Text } from "@chakra-ui/react"
+import type { IconType } from "react-icons/lib"
+import Button, { type IProps as ButtonProps } from "../Button"
+import ButtonLink, { type IProps as ButtonLinkProps } from "../ButtonLink"
+
+type CommonProps = {
+  icon: IconType | typeof Icon
+  iconAlignment?: "left" | "right"
+  /**
+   * Reduced choices of the button variant.
+   *
+   * This component only accepts the `solid` or `outline` variant
+   */
+  variant?: "solid" | "outline"
+  /**
+   * Reduced choices of the button size
+   *
+   * This component only accepts the `md` or `sm` sizes
+   */
+  size?: "md" | "sm"
+  mainText: string
+  helperText: string
+  /**
+   * Should the main text be below the helper text instead of ab?
+   */
+  reverseTextOrder?: boolean
+}
+
+type OmittedTypes = "variant" | "size"
+
+type ButtonTypeProps = CommonProps &
+  Omit<ButtonProps, OmittedTypes> & {
+    href?: never
+  }
+
+type ButtonLinkTypeProps = CommonProps &
+  Omit<ButtonLinkProps, OmittedTypes> & {
+    toId?: never
+  }
+
+type ButtonTwoLinesProps = ButtonTypeProps | ButtonLinkTypeProps
+
+const hasHref = (props: ButtonTwoLinesProps): props is ButtonLinkTypeProps => {
+  return "href" in props
+}
+
+const ButtonTwoLines = (props: ButtonTwoLinesProps) => {
+  const {
+    icon: Icon,
+    iconAlignment = "left",
+    mainText,
+    helperText,
+    reverseTextOrder = false,
+    size = "md",
+    ...rest
+  } = props
+
+  const isIconLeft = iconAlignment === "left"
+
+  const vertPadding: ButtonTwoLinesProps["py"] = size === "md" ? "4" : "2"
+
+  const buttonStyles: ButtonProps = {
+    [isIconLeft ? "leftIcon" : "rightIcon"]: <Icon />,
+    textAlign: isIconLeft ? "start" : "end",
+    justifyContent: isIconLeft ? "flex-start" : "flex-end",
+    w: "278px",
```

Comment:
Probably we've talked about this value but could you remind me why we want this fixed width?

Reply:
I believe we have yet to discuss this. 😁

I had noticed at one point in the figma file some playing around with the text length and having it wrap and keep the same button length.

With the expected usage, I would think having a fixed width to maintain uniformity for every instance of this component.

Alternatively, could make it `min-width` instead and allow text to stretch the button and not wrap.

Reply:
Interesting. `min-width` would make more sense to me here.

@nloureiro is this something we want to keep in code? or was just to display it nicely in figma?